packages/react-kit/src/components/Header/Header.tsx

```diff
@@ -15,34 +15,29 @@ export const Header = ({
 	navbarStyles,
 	focusTrapOptions,
 	linkComponent,
+	mobileNavHandler: [isOpen, handleOpen],
 }: NavHeaderProps) => {
-	const [isOpen, setIsOpen] = useState(false);
-	const [windowWidth, setWindowWidth] = useState<number>(0);
-	const handleOpen = () => setIsOpen((prev) => !prev);
-
-	const MAX_MOBILE_NAV_WIDTH = 1023;
 	const LinkComponent = linkComponent ? linkComponent : 'a';
 
 	useEffect(() => {
 		const handleKeyUp = (e: KeyboardEvent) => {
 			if (e.key === 'Escape' || e.key === 'Esc') {
-				setIsOpen(false);
+				handleOpen();
 			}
 		};
 		const handleResize = () => {
-			setWindowWidth(window?.innerWidth);
-			windowWidth > MAX_MOBILE_NAV_WIDTH && setIsOpen(false);
+			if (window.innerWidth >= 1024 && isOpen) {
+				handleOpen();
+			}
 		};
-		setWindowWidth(window?.innerWidth);
-
-		window?.addEventListener('resize', handleResize);
-		window?.addEventListener('keyup', handleKeyUp);
+		window.addEventListener('keyup', handleKeyUp);
+		window.addEventListener('resize', handleResize);
 
 		return () => {
-			window?.removeEventListener('resize', handleResize);
-			window?.removeEventListener('keyup', handleKeyUp);
+			window.removeEventListener('keyup', handleKeyUp);
+			window.removeEventListener('resize', handleResize);
 		};
-	}, [windowWidth]);
+	}, [isOpen, handleOpen]);
```

Comment:
@backlineint  In either case, if I leave the dependency array empty the linter yells at me. If you could show an example or maybe we could sync on this, would be helpful.

Reply:
Pesky linters.

Given that, the closest thing I could come up with would be to move the function definitions outside of useEffect and wrap them in useCallback so they can be cached. Something like:

```
	const handleKeyUp = useCallback(
		(e: KeyboardEvent) => {
			if (e.key === 'Escape' || e.key === 'Esc') {
				handleOpen();
			}
		},
		[handleOpen],
	);
	const handleResize = useCallback(() => {
		if (window.innerWidth >= 1024 && isOpen) {
			handleOpen();
		}
	}, [handleOpen, isOpen]);

	useEffect(() => {
		window.addEventListener('keyup', handleKeyUp);
		window.addEventListener('resize', handleResize);

		return () => {
			window.removeEventListener('keyup', handleKeyUp);
			window.removeEventListener('resize', handleResize);
		};
	}, [handleKeyUp, handleResize]);
```

If I'm parsing about this correctly I think that would be slightly better - the handleKeyUp dependency would never change, but handleResize would still trigger useEffect when the isOpen state changes. I don't think that is worth the effort, especially since the two layers of dependencies makes this harder to reason about.

So all that to say, let's leave it as is :) The effect does remove the event listener, so it seems unlikely that it would result in an actual performance problem.